lib/types.ts

```diff
@@ -156,6 +156,8 @@ export type DeepReadonly<T> = T extends Builtin
   ? Promise<DeepReadonly<U>>
   : T extends {}
   ? { readonly [K in keyof T]: DeepReadonly<T[K]> }
+  : T extends unknown
```

Comment:
Also, instead of `T extends unknown` just in case, let's use `unknown extends T`

Example here – https://tsplay.dev/Nd4vXN